src/modules/account/views/ViewAccountFailedTransactions.vue

```diff
@@ -52,12 +81,34 @@ const checkTxnStatus = async () => {
 
     const currentAccountPubKey = currentAccount? currentAccount.publicKey: ""
 
-    let activityFailedTxn = AppState.txnActivityLog
-        .filter((x) => x.status === "failed" && x.accPubKey === currentAccountPubKey)
-    let cosignFailedTxn = AppState.txnCosignLog
-        .filter((x) => x.status === "failed" && x.accPubKey.includes(currentAccountPubKey))
-    let swapFailedTxn = AppState.txnSwapLog
-        .filter((x) =>  x.status === "failed" && x.accPubKey === currentAccountPubKey)
+    if(AppState.txnActivityLog.length>0){
+        let fetchFailedTxn = JSON.parse(sessionStorage.getItem("txnFailedActivityLog")) || []
+        let concatFailedTxns = fetchFailedTxn.concat(AppState.txnActivityLog.filter((x) => x.status === "failed"))
+        let filterFailedTxns = concatFailedTxns.filter((value: { txnHash: string; }, index: number) => index === concatFailedTxns.findIndex((t: { txnHash: string; }) => (t.txnHash === value.txnHash)))
+        sessionStorage.setItem("txnFailedActivityLog", JSON.stringify(filterFailedTxns))
+    }
+    if(AppState.txnCosignLog.length>0){
+        let fetchFailedTxn = JSON.parse(sessionStorage.getItem("txnFailedCosignLog")) || []
+        let concatFailedTxns = fetchFailedTxn.concat(AppState.txnCosignLog.filter((x) => x.status === "failed"))
+        let filterFailedTxns = concatFailedTxns.filter((value: { txnHash: string; }, index: number) => index === concatFailedTxns.findIndex((t: { txnHash: string; }) => (t.txnHash === value.txnHash)))
+        sessionStorage.setItem("txnFailedCosignLog", JSON.stringify(filterFailedTxns))
+    }
+    if(AppState.txnSwapLog.length>0){
+        let fetchFailedTxn = JSON.parse(sessionStorage.getItem("txnFailedSwapLog")) || []
+        let concatFailedTxns = fetchFailedTxn.concat(AppState.txnSwapLog.filter((x) => x.status === "failed"))
+        let filterFailedTxns = concatFailedTxns.filter((value: { txnHash: string; }, index: number) => index === concatFailedTxns.findIndex((t: { txnHash: string; }) => (t.txnHash === value.txnHash)))
+        sessionStorage.setItem("txnFailedSwapLog", JSON.stringify(filterFailedTxns))
+    }
+
+    let storeActivityFailedTxn :TxnActivityLog[] = JSON.parse(sessionStorage.getItem("txnFailedActivityLog")) || []
+    let storeCosignFailedTxn :TxnCosignLog[] = JSON.parse(sessionStorage.getItem("txnFailedCosignLog")) || []
+    let storeSwapFailedTxn :TxnSwapLog[] = JSON.parse(sessionStorage.getItem("txnFailedSwapLog")) || []
+
+    let activityFailedTxn = storeActivityFailedTxn.filter((x: { accPubKey: string; }) => x.accPubKey === currentAccountPubKey)
+
+    let cosignFailedTxn = storeCosignFailedTxn.filter((x: { accPubKey: string | string[]; }) => x.accPubKey.includes(currentAccountPubKey))
+
+    let swapFailedTxn = storeSwapFailedTxn.filter((x: { accPubKey: string; }) =>  x.accPubKey === currentAccountPubKey)
```

Comment:
no need to concat, just read from session storage and filter with public key. You only store into session storage when those txn become failed (header file)